Case study (#4158)
layer5io/layer5: src/collections/resources/articles/hpes-adoption-of-meshery-and-meshmap/index.mdx

---
title: HPE's adoption of Meshery and Meshmap
subtitle:
date: 2023-04-28 08:00:00 -0530
author: HPE
thumbnail: ./meshery-and-hpe.svg
darkthumbnail: ./meshery-and-hpe.svg
description: "HPE uses Meshery to manage SPIRE instances"
type: Case Study
category: Cloud Native
tags:
  - Cloud Native, Meshery
featured: false
published: true
resource: true
---

import { ResourcesWrapper } from "../../Resources.style.js";
import { HPEfacts, HPEintro } from "./hpe.style.js";
import { theme } from "../../../../theme/app/themeStyles.js";

import { Link } from "gatsby";
import Button from "../../../../reusecore/Button";
import GlobeIcon from "./globe.svg";
import UsersIcon from "./users.svg";
import NetworkIcon from "./network.svg";
import LayersIcon from "./layers.svg"

<ResourceWrapper>

<HPEintro>

>    HPE's adoption of Meshery was driven by the need to simplify the management and monitoring of their Kubernetes clusters. Meshery is an open source cloud native management tool that provides a self-service platform for designing, deploying, testing, and operating cloud native infrastructure wherever Kubernetes and its workloads run.

Reviewer: ```suggestion
HPE's adoption of Meshery was driven by the need to simplify Kubernetes cluster management and monitoring. Meshery is an open source cloud native management tool that provides a self-service platform for designing, visualizing, deploying, testing, and operating cloud native infrastructure.
```

</HPEintro>

<p>

>    HPE, a leading technology company specializing in enterprise infrastructure, adopted Layer5 MeshMap to enhance their Kubernetes deployments. HPE uses Kubernetes as a primary platform to build and deploy their containerized applications. The company has a large and complex Kubernetes environment, which requires robust networking solutions for efficient communication between services.

Reviewer: ```suggestion
   HPE, a leading technology company specializing in enterprise infrastructure, adopted Meshery Extension to enhance their Kubernetes deployments. HPE uses Kubernetes as a primary platform to build and deploy their containerized applications. The company has a large and complex Kubernetes environment, which requires robust networking solutions for efficient communication between services.
```

</p>

<HPEfacts>
   <tr><td colspan="2"><h4>HPE FAST FACTS</h4></td></tr>
   <tr>
      <td>
      <img src={UsersIcon} />
      Full Time Employees: 60,000+
      </td>
      <td>
      <img src={GlobeIcon} />
      Market Presence: HPE is one of the largest technology companies globally, serving customers in over 150 countries.
      </td>
   </tr>
   <tr>
      <td>
      <img src={LayersIcon} />
      HPE GreenLake: HPE GreenLake is a key offering by the company, providing a flexible and scalable IT infrastructure model known as "everything-as-a-service."
      </td>
      <td>
      <img src={NetworkIcon} />
      Research and Development: HPE invests significantly in research and development to drive innovation. It operates HPE Labs, which focuses on developing cutting-edge technologies and solutions for the future.
      </td>
   </tr>
</HPEfacts>

<p>

> SPIRE is a toolchain of APIs for establishing trust between software systems across a wide variety of hosting platforms. SPIRE exposes the SPIFFE Workload API, which can attest running software systems and issue SPIFFE IDs and SVIDs to them. <br/><br/>

Reviewer: ```suggestion
SPIRE is a toolchain of APIs for establishing trust between software systems across a wide variety of hosting platforms. SPIRE exposes the SPIFFE Workload API, which attests to running software systems and issues SPIFFE IDs and SVIDs to them.<br/><br/>
```

Reviewer 2: The current grammar is correct.

    <b>Meshery offers several benefits to HPE, including:</b>
    <li>
    Consistent service mesh management: HPE can now manage all their service meshes consistently, regardless of the underlying infrastructure or cloud provider.
    </li>
    <li>
    Improved observability: With Meshery's built-in visualization and observability tools, HPE can gain insights into the behavior of their service meshes, detect anomalies, and troubleshoot issues in real-time.
    </li>
    <li>
    Simplified testing and validation: Meshery's service mesh validation capabilities enable HPE to easily test and validate their service meshes, ensuring that they meet their performance, security, and compliance requirements.
    </li>
    <li>
    Enhanced security: With Meshery's security features, HPE can ensure that their service meshes are secure and compliant with their organization's security policies.
    </li>
</p>

<p>

>     Overall, Meshery has helped HPE to streamline their integration of the identity management control plane to reduce complexity, and improve the overall reliability and performance of their Kubernetes environment. SPIFFE is a set of open-source specifications for a framework capable of bootstrapping and issuing identity to services across heterogeneous environments and organizational boundaries. The lifecycle of SPIFFE identities, SVIDs, is managed by SPIRE, a production-ready implementation of the SPIFFE APIs that performs node and workload attestation in order to securely issue SVIDs to workloads, and verify the SVIDs of other workloads, based on a predefined set of conditions.

Reviewer: ```suggestion
Overall, Meshery has helped HPE to streamline their integration of the identity management control plane to reduce complexity, and improve the overall reliability and performance of their Kubernetes environment. SPIFFE is a set of open-source specifications for a framework capable of bootstrapping and identifying services across heterogeneous environments and organizational boundaries. The lifecycle of SPIFFE identities, SVIDs, is managed by SPIRE, a production-ready implementation of the SPIFFE APIs that performs node and workload attestation to securely issue SVIDs to workloads, and verify the SVIDs of other workloads, based on a predefined set of conditions.
```

</p>

<p>
   SMP is a community-driven effort that provides a standard for measuring and comparing the performance of different service meshes. It is designed to help users select the best service mesh for their needs by providing a common framework for benchmarking.
</p>

<p>
   Meshery implements SMP by providing a simple and easy-to-use interface for running performance tests against different service meshes. Users can select the service mesh they want to test, configure the test parameters (such as the number of requests per second and the number of concurrent clients), and run the test. Meshery will then generate a report that shows the performance metrics for each service mesh, such as latency, throughput, and error rates.
</p>

<p>
   By implementing SMP, Meshery provides a valuable tool for developers and operators who are evaluating different service meshes. Instead of having to create their own benchmarks, they can use SMP to get an objective and standardized view of each service mesh's performance characteristics. This can save a significant amount of time and effort, and help users make more informed decisions when choosing a service mesh.
</p>

<p>

> Overall, HPE's use of Meshery and the Docker Extension for Meshery demonstrates the power of cloud native technologies and the importance of open source collaboration. By leveraging these tools, HPE has been able to streamline its development and deployment processes, improve performance and security, and stay at the forefront of the cloud native movement.

Reviewer: Let's add some quotes from HPE folks here if we could.

Author: @Chadha93 @leecalcote we can use the quote from Maximiliano here.

Reviewer 2: Sounds great. I'll send another from another HPE employeee.
